### PR TITLE
Rename icu_segmenter enums based on feedback

### PR DESCRIPTION
**Pull request by sffc**
@markusicu made suggestions for new names of the enums in the icu_segmenter component in an API review.

### CODE REVIEW

**Review by eggrobin — APPROVED** [review]

File: experimental/segmenter/src/word.rs [hunk 1]
```diff
@@ -41,11 +41,28 @@ pub struct WordBreakIterator<'l, 's, Y: RuleBreakType<'l, 's> + ?Sized>(
 
 derive_usize_iterator_with_type!(WordBreakIterator);
 
+/// The word type derived from the rule status in a rule-based word segmenter.
```
Comment (aethanyc) [inline experimental/segmenter/src/word.rs:44]: Suggestion: this is public doc, so describing an implementation detail is probably not very useful. How about something like:  ``/// The word type tag that is returned by [`WordBreakIterator::word_type()`].``

File: experimental/segmenter/src/word.rs [hunk 2]
```diff
@@ -41,11 +41,28 @@ pub struct WordBreakIterator<'l, 's, Y: RuleBreakType<'l, 's> + ?Sized>(
...
+    /// No category tag
+    None = 0,
+    /// Number category tag
+    Number = 1,
+    /// Letter category tag, including CJK.
+    Letter = 2,
+}
+
 impl<'l, 's, Y: RuleBreakType<'l, 's> + ?Sized> WordBreakIterator<'l, 's, Y> {
     /// Return the status value of break boundary.
```
Comment (aethanyc) [inline experimental/segmenter/src/word.rs:58]: Suggestion: In #3281, I've changed this line. With you change, we should revise this. Maybe something like
"/// Return the word type tag value of the current word break boundary."

File: experimental/segmenter/src/word.rs [hunk 3]
```diff
@@ -41,11 +41,28 @@ pub struct WordBreakIterator<'l, 's, Y: RuleBreakType<'l, 's> + ?Sized>(
 
 derive_usize_iterator_with_type!(WordBreakIterator);
 
+/// The word type derived from the rule status in a rule-based word segmenter.
+#[non_exhaustive]
+#[derive(Copy, Clone, PartialEq, Debug)]
+#[repr(u8)]
+pub enum WordType {
+    /// No category tag
```
Comment (aethanyc) [inline experimental/segmenter/src/word.rs:49]: Nit: add period at the end of the sentence.

File: experimental/segmenter/src/word.rs [hunk 4]
```diff
@@ -41,11 +41,28 @@ pub struct WordBreakIterator<'l, 's, Y: RuleBreakType<'l, 's> + ?Sized>(
...
+    None = 0,
+    /// Number category tag
```
Comment (aethanyc) [inline experimental/segmenter/src/word.rs:51]: Nit: add period at the end of the sentence.

**Review by aethanyc — APPROVED** [review]
The rename looks good.

praise: I like this PR that makes `rule_segmenter` an implementation detail.

File: experimental/segmenter/src/word.rs (hunk 2 above)
Comment (sffc) [reply experimental/segmenter/src/word.rs:58]: How about (as suggested by @markusicu):

```
/// Returns the word type of the segment preceding the current boundary.
```

Comment (aethanyc) [reply experimental/segmenter/src/word.rs]: This is better.

**Pull request merged**